pyo3-macros-backend/src/pyclass.rs

```diff
@@ -230,11 +241,33 @@ pub fn build_py_class(
             })
             .collect::<Result<_>>()?,
         syn::Fields::Unit => {
+            if let Some(attr) = args.options.set_all {
+                return Err(syn::Error::new(attr.span(), UNIT_SET));
+            };
+            if let Some(attr) = args.options.get_all {
+                return Err(syn::Error::new(attr.span(), UNIT_GET));
+            };
             // No fields for unit struct
             Vec::new()
         }
     };
 
+    if let Some(attr) = args.options.get_all {
+        for (_, FieldPyO3Options { get, .. }) in &mut field_options {
+            if mem::replace(get, true) {
+                return Err(syn::Error::new(attr.span(), DUPE_GET));
+            }
+        }
+    }
+
+    if let Some(attr) = args.options.set_all {
+        for (_, FieldPyO3Options { set, .. }) in &mut field_options {
+            if mem::replace(set, true) {
+                return Err(syn::Error::new(attr.span(), DUPE_SET));
+            }
+        }
```

Comment:
IMO the spans for the errors here aren't ideal because they use the span of `get_all` and `set_all`, but the error is describing the `get` and `set` as the source of the error.

Using the `syn::Field` span (bound as `_` in these snippets) may be better as it'll point to the field which has the invalid attribute?

The ideal span would probably be the `get` attribute itself - you could convert the `FieldPyO3Options` to have `get: Option<kw::get>`, and then use those spans. It would however mean that your elegant solution of using `mem::replace` here wouldn't be quite enough.

Reply:
That's a good idea, done.